packages/arb-ts/src/lib/l1Bridge.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019-2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-env node */
+'use strict'
+import { providers, Signer, BigNumber } from 'ethers'
+import { InboxFactory } from './abi/InboxFactory'
+import { EthERC20BridgeFactory } from './abi/EthERC20BridgeFactory'
+import { EthERC20Bridge } from './abi/EthERC20Bridge'
+import { Inbox } from './abi/Inbox'
+
+export class L1Bridge {
+  l1Provider: providers.JsonRpcProvider
+  l1Signer: Signer
+  inbox: Inbox
+  ethERC20Bridge: EthERC20Bridge
+  walletAddressCache?: string
+
+  constructor(
+    inboxAddress: string,
+    erc20BridgeAddress: string,
+    l1Provider: providers.JsonRpcProvider,
+    l1Signer: Signer
+  ) {
+    this.l1Provider = l1Provider
+    this.l1Signer = l1Signer
+
+    const l1SignerOrProvider = l1Signer || l1Provider
+    this.inbox =
+      l1SignerOrProvider &&
+      InboxFactory.connect(inboxAddress, l1SignerOrProvider)
+
+    this.ethERC20Bridge = EthERC20BridgeFactory.connect(
+      erc20BridgeAddress,
+      l1Signer
+    )
+  }
+
+  public async depositETH(value: BigNumber, destinationAddress?: string) {
+    if (!this.l1Signer || !this.inbox) {
+      throw new Error('No eth signer provider')
+    }
+    const address = destinationAddress || (await this.getWalletAddress())
+    return this.inbox.depositEth(address, {
+      value,
+    })
+  }
+
+  public async depositERC20(
+    erc20L1Address: string,
+    amount: BigNumber,
+    maxGas: BigNumber,
+    gasPriceBid: BigNumber,
+    destinationAddress?: string
+  ) {
+    const destination = destinationAddress || (await this.getWalletAddress())
+    this.ethERC20Bridge.depositAsERC20(
+      erc20L1Address,
+      destination,
+      amount,
+      maxGas,
+      gasPriceBid
+    )
+  }
+
+  public async getERC20L2Address(erc20L1Address: string) {
+    return await this.ethERC20Bridge.customL2Tokens(erc20L1Address)
```

Comment:
We should check if there is a customL2Token that was assigned, if not then we can calculate the create2 address for erc20. 

```
const l2Address = ethers.utils.getCreate2Address(
      bridge.address,
      ethers.utils.hexZeroPad(l1ERC20, 32),
      ethers.utils.keccak256(proxyBytecode)
)

Reply:
true, tho the likely intended use is to interact with it; it should maybe be more of a 'get address if contract is deployed' sort of method. idk